onnxruntime/python/tools/transformers/models/whisper/benchmark.py

```diff
@@ -0,0 +1,696 @@
+    logger.info(f"Batch size = {metrics[0]}, latency = {metrics[1]} s, throughput = {metrics[2]} qps")
+
+
+# Benchmark types: ORT only
+def run_ort_only_inference(args, inputs, model):
+    # Check that all model inputs will be provided
+    model_inputs = set(map(lambda model_input: model_input.name, model.get_inputs()))
+    user_inputs = set(inputs.keys())
+    missing_inputs = model_inputs - user_inputs
+    if len(missing_inputs):
+        logger.error(f"The following model inputs are missing: {missing_inputs}")
+        raise Exception("There are missing inputs to the model. Please add them to `get_ort_inputs`.")
+
+    # Remove unnecessary inputs from model inputs
+    unnecessary_inputs = user_inputs - model_inputs
+    if len(unnecessary_inputs):
+        for unnecessary_input in unnecessary_inputs:
+            logger.info(f"Removing unnecessary input '{unnecessary_input}' from user provided inputs")
+            del inputs[unnecessary_input]
+
+    # Add IO binding for non-CPU inputs
+    if args.device != "cpu":
+        io_binding = model.io_binding()
+        for k, v in inputs.items():
+            io_binding.bind_cpu_input(k, v)
+        for output in model.get_outputs():
+            io_binding.bind_output(output.name)
+
+    if args.profile:
+        # Measure CPU usage
+        pid = os.getpid()
+        process = psutil.Process(pid)
+        process.cpu_percent(interval=0.1)
+
+        model.run(None, inputs) if args.device == "cpu" else model.run_with_iobinding(io_binding)
+        logger.info(f"CPU percentage = {process.cpu_percent(interval=None)}%")
+
+        # Turn profiling off to stop generating logs
+        args.profile = False
+        model.end_profiling()
+
+        # Measure memory usage
+        gc.collect()
+        torch.cuda.empty_cache()
+        measure_memory(is_gpu=(args.device == "cuda"), func=lambda: model.run(None, inputs))
+
+        return
+
+    if args.device == "cpu":
+        generate_fn = lambda inputs: model.run(None, inputs)  # noqa: E731
+        outputs, metrics = time_fn(args, generate_fn, inputs)
+    else:
+        generate_fn = lambda io_binding: model.run_with_iobinding(io_binding)  # noqa: E731
+        outputs, metrics = time_fn(args, generate_fn, io_binding)
+    logger.info(f"Batch size = {metrics[0]}, latency = {metrics[1]} s, throughput = {metrics[2]} qps")
+
+
+def run_inference(args, inputs, processor, model, pipeline):
+    if args.hf_api == "pipeline":
+        run_hf_pipeline_inference(args, inputs, pipeline)
+    elif args.hf_api == "gen-and-dec":
+        run_hf_generate_and_decode_inference(args, inputs, processor, model)
+    else:
+        run_ort_only_inference(args, inputs, model)
+
+
+def parse_args():
+    parser = argparse.ArgumentParser()
+
+    # Args for benchmark type
+    parser.add_argument(
+        "-bt", "--benchmark-type", type=str, required=True, choices=["HF + PT", "HF + PT2", "HF + ORT", "ORT"]
```

Comment:
Use one word like hf_pt, hf_pt2, hf_ort, ort. May add some description about what the meaning of each is.